src/eradiate/scenes/atmosphere/_core.py

```diff
@@ -11,22 +11,123 @@
 import pinttr
 import xarray as xr
 
-from ..core import BoundingBox, KernelDict, SceneElement
+from ..core import KernelDict, SceneElement
+from ..shapes import CuboidShape, SphereShape
 from ... import converters
 from ... import unit_context_kernel as uck
 from ... import validators
 from ..._factory import Factory
 from ..._util import onedict_value
-from ...attrs import AUTO, documented, get_doc, parse_docs
+from ...attrs import AUTO, AutoType, documented, get_doc, parse_docs
 from ...contexts import KernelDictContext, SpectralContext
 from ...kernel.gridvolume import write_binary_grid3d
-from ...kernel.transform import map_cube, map_unit_cube
+from ...kernel.transform import map_unit_cube
 from ...units import to_quantity
 from ...units import unit_context_config as ucc
+from ...units import unit_registry as ureg
 
 atmosphere_factory = Factory()
 
 
+@parse_docs
+@attr.s
+class AtmosphereGeometry:
+    """
+    Base class defining the geometry of the atmosphere.
+    """
+
+    @classmethod
+    def convert(cls, value: t.Any) -> t.Any:
+        """
+        Attempt conversion of a value to a :class:`AtmosphereGeometry` subtype.
+
+        Parameter
+        ---------
+        value
+            Value to attempt conversion of. If a dictionary is passed, its
+            ``"type"`` key is used to route its other entries as keyword
+            arguments to the appropriate subtype's constructor. If a string is
+            passed, this method calls itself with the parameter
+            ``{"type": value}``.
+
+        Returns
+        -------
+        If `value` is a dictionary, the constructed :class:`AtmosphereGeometry`
+        instance is returned. Otherwise, `value` is returned.
+
+        Raises
+        ------
+        ValueError
+            A dictionary was passed but the requested type is unknown.
+        """
+        if isinstance(value, str):
+            return cls.convert({"type": value})
+
+        if isinstance(value, dict):
+            value = value.copy()
+            geometry_type = value.pop("type")
+
+            try:
+                geometry_cls = {
+                    "plane_parallel": PlaneParallelGeometry,
+                    "spherical_shell": SphericalShellGeometry,
+                }[geometry_type]
+            except KeyError:
+                raise ValueError(f"unknown geometry type '{geometry_type}'")
```

Comment:
Using a mapping is the optimal equivalent to a switch statement in Python < 3.10. Access to mapping elements has constant complexity, while the conditional structure has linear complexity.

Arguably, in that specific case, it doesn't make much difference; it's actually probably not that good since there is a dict creation at each execution. Doing it the right way would actually require a factory: it's clearly overkill. I'll optimise this and write a conditional structure.